File: 06 arrays/exercise - maxSubArray.py
# 53. Maximum Subarray
# https://leetcode.com/problems/maximum-subarray/description/

class Solution(object):
  def maxSubArray(self, nums):
    if (not nums):
      raise ValueError("invalid input")

    maxSubArray = 0

    for i in range(len(nums)):
      sum = 0
      for j in range(i, len(nums)):
        sum += nums[j]
        maxSubArray = max(maxSubArray, sum)

    return maxSubArray

File: 06 arrays/exercise - maxSubArray.js
// 53. Maximum Subarray
// https://leetcode.com/problems/maximum-subarray/description/

var maxSubArray = function (nums) {
  if (!nums)
    throw "invalid input";
     
  let maxSubArray = 0;

  for (let i = 0; i < nums.length; i++) {
    let sum = 0;
    for (let j = i; j < nums.length; j++) {
      sum += nums[j];
      maxSubArray = Math.max(maxSubArray, sum);
    }
  }

  return maxSubArray;
}
